girder/api/access.py

```diff
@@ -80,3 +77,15 @@ def accessDecorator(*args, **kwargs):
         return fun(*args, **kwargs)
     accessDecorator.accessLevel = 'token'
     return accessDecorator
+
+
+def cookie(fun):
+    """
+    REST endpoints that allow the use of a cookie for authentication should be
+    wrapped in this decorator.
+
+    This should only be set on endpoints that are read-only (GET requests),
+    to avoid vulnerabilities to Cross-Site Request Forgery attacks.
```

Comment:
The security-sensitive part of me wants to restrict this to not work on anything but a GET request unless explicitly stated through a parameter to the decorator...